### ISSUE TITLE
[Javascript] Implement new Concept Exercise: callbacks

### ISSUE THREAD

**Issue by rishiosaur** [exercism/v3#1458]
This issue describes how to implement the `callbacks` concept exercise for the Javascript track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

Javascript prides itself on being a functional-paradigm language. One of the main features of any functional language is the ability to pass around functions as first-class objects. This lesson will teach students how to use functions as parameters to existing 

## Learning objectives

- Know how to use functions as parameters in other functions
- Use callbacks in asynchronous code
- Node error messages (thank you for the idea @SleeplessByte)

## Out of scope

- Closures
- Currying
- "Private" methods & properties

## Concepts

- `callbacks-basic`

## Prerequisites

- `functions`
- `booleans`

## Resources to refer to
- [MDN Callbacks](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function)
- [Asynchronous Callbacks](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Introducing#Async_callbacks)

### Hints

- [How to run functions from parameters](https://www.dashingd3js.com/lessons/javascript-callback-functions)

### After

- TODO

## Representer

- N/A

## Analyzer

- N/A

## Implementing

- N/A

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by SleeplessByte** [comment]
So the concept is called `callbacks`, not `callback` or `callbacks-basic`, and it should probably _not_ depend on `promises`. 

I don't know yet how we would do `functions`, but I think `closures`/`higher-order-functions`/`first-class-functions` (that would be the Concept) can be modelled after [`elixir/anonymous-functions`](https://github.com/exercism/v3/blob/master/languages/elixir/exercises/concept/anonymous-functions/.docs/instructions.md).

I definitely agree with your Out of Scope :).

- Know how to use functions as parameters in other functions
- ~Use callbacks in asynchronous code~
- Call a `callback` conditionally
- (MAYBE) node-style error callbacks (so `callback(err, data)`)

**Comment by SleeplessByte** [comment]
@rishiosaur any more questions after my last message?

**Comment by neenjaw** [comment]
@rishiosaur are you working on this exercise?

**Comment by rishiosaur** [comment]
@neenjaw go for it!

**Comment by neenjaw** [comment]
In regard to this concept.  Asynchronous callbacks are often in the context of browser events (`addEventListener`) and in nodejs api calls like `fs` and `fs.read`.

Which context is the student working in, DOM or node?

Then in trying to think about the exercise, writing a function for a synchronous call back should be easy, but any suggestions for scope or focus for an async callback? Is is async callbacks like event handlers out of the scope of this exercise?

**Comment by shubhsk88** [comment]
@neenjaw  Are you still working on this issue?I would  like to contribute to it

**Comment by SleeplessByte** [comment]
@shubhsk88 yeah, @neenjaw and I talked on slack about it. It's basically on hold until we solve some other thangs.

**Comment by SleeplessByte** [comment]
Closed by #2266
